imap_processing/idex/IDEXPacketParser.py

```diff
@@ -0,0 +1,120 @@
+import bitstring
+import numpy as np
+from space_packet_parser import parser, xtcedef
+from imap_processing import packet_definition_directory
+
+
+class IDEXPacketParser():
+    def __init__(self, idex_packet_file: str):
+
+        idex_xtce = f"{packet_definition_directory}/idex_packet_definition.xml"
+        idex_definition = xtcedef.XtcePacketDefinition(xtce_document=idex_xtce)
+        idex_parser = parser.PacketParser(idex_definition)
+
+        idex_binary_data = bitstring.ConstBitStream(filename=idex_packet_file)
+        idex_packet_generator = idex_parser.generator(idex_binary_data)
+
+        self.epochs = []
+        self.data = {}
+        
+        evtnum = 0
+        for pkt in idex_packet_generator:
+            if 'IDX__SCI0TYPE' in pkt.data:
```

Comment:
This isn't really a comment to this specific PR, but what are the chances of packet field names changing? If there's a possibility that they could change down the line would it make sense to create some type of adapter so we wouldn't need to go through and change these strings? Or if they did change would we not care since we can call them whatever we want in the xtce files?

Reply:
So I'm certainly not an expert at decom or XTCE, but I think we're just assigning a human-readable name to those specific 8 bits in the file.  I went through and changed the code and the packet definition xml file to call it 'IDX__SCI0TYPE_X' instead of "IDX__SCI0TYPE", and everything still decoded fine.